dev/com.ibm.ws.security.wim.adapter.ldap_fat.krb5.1/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/krb5/KerberosContainer.java

```diff
@@ -95,38 +98,34 @@ protected void configure() {
                 Log.info(c, "configure", "ExposedPort=" + p.getPort());
                 exposedPorts.add(p);
             }
-            exposedPorts.add(ExposedPort.tcp(88));
+            //exposedPorts.add(ExposedPort.tcp(88));
+            exposedPorts.add(ExposedPort.udp(99));
             cmd.withExposedPorts(exposedPorts);
 
             // Add previous port bindings and KDC and LDAP ports
             Ports ports = cmd.getPortBindings();
-            int containerPort = 88;
-            int hostPort = 88;
-
-            String kdcPortMapping = String.format("%d:%d/%s", hostPort, containerPort, InternetProtocol.TCP);
-            Log.info(c, "configure", "adding KDC port mapping: " + kdcPortMapping);
-
-            Log.info(c, "configure", "PortBinding.parse(kdcPortMapping): " + PortBinding.parse(kdcPortMapping));
-            ports.add(PortBinding.parse(kdcPortMapping));
+            ports.bind(ExposedPort.udp(99), Ports.Binding.empty());
 
-            Log.info(c, "configure", "ports: " + ports);
+            //Log.info(c, "configure", "ports: " + ports);
             cmd.withPortBindings(ports);
 
-            cmd.withHostName(KRB5_KDC);
+            cmd.withHostName(KRB5_KDC_EXTERNAL);
         });
     }
 
     @Override
     protected void containerIsStarted(InspectContainerResponse containerInfo) {
-        String udp88 = containerInfo.getNetworkSettings().getPorts().getBindings().get(new ExposedPort(88, InternetProtocol.TCP))[0].getHostPortSpec();
-        tcp_88 = Integer.valueOf(udp88);
+        String udp88 = containerInfo.getNetworkSettings().getPorts().getBindings().get(new ExposedPort(99, InternetProtocol.UDP))[0].getHostPortSpec();
+        udp_99 = Integer.valueOf(udp88);
+        KRB5_PORT = "" + udp_99;
+        Log.info(c, "containerIsStarted", "Using KRB5_PORT=" + KRB5_PORT);
     }
 
     @Override
     public Integer getMappedPort(int originalPort) {
         // For this container assume we always want the UDP port when we ask for port 99
         if (originalPort == 88) {
```

Comment:
```suggestion
        if (originalPort == 99) {
```
Note: Might require refactoring in other places, but will cause less confusion later on to fix this now.